python/cudf/cudf/core/dataframe.py

```diff
@@ -104,6 +103,11 @@
     _external_only_api,
 )
 
+if sys.version_info < (3, 10):
```

Comment:
Rather than hard-code the version, try to import it and define the fallback in an exception handler.

```python
try:
    # Introduced in Python 3.10
    from types import NotImplementedType
except ImportError:
    NotImplementedType = Any
```

Reply:
Unfortunately that doesn't work (because `ImportError` is not a statically determinable condition):

In [3.8 or 3.9](https://mypy-play.net/?mypy=latest&python=3.8&gist=181adc25354058b0dc50790c1934a78f)
```
main.py:3: error: Module "types" has no attribute "NotImplementedType"
Found 1 error in 1 file (checked 1 source file)
```

In [3.10 or 3.11](https://mypy-play.net/?mypy=latest&python=3.10&gist=181adc25354058b0dc50790c1934a78f)
```
main.py:5: error: Cannot assign multiple types to name "NotImplementedType" without an explicit "Type[...]" annotation
main.py:5: error: Incompatible types in assignment (expression has type "object", variable has type "Type[_NotImplementedType]")
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
Wow! I had not anticipated that static condition. That’s interesting.